gcc/rust/typecheck/rust-hir-type-check-path.cc

```diff
@@ -218,8 +218,17 @@ TypeCheckExpr::resolve_root_path (HIR::PathInExpression &expr, size_t *offset,
 	    = Resolver2_0::ImmutableNameResolutionContext::get ().resolver ();
 
 	  // assign the ref_node_id if we've found something
-	  nr_ctx.lookup (expr.get_mappings ().get_nodeid ())
-	    .map ([&ref_node_id] (NodeId resolved) { ref_node_id = resolved; });
+	  nr_ctx.lookup (ast_node_id).map ([&ref_node_id] (NodeId resolved) {
+	    ref_node_id = resolved;
+	  });
+
+	  // HACK: on failure, redo lookup using path node id
+	  // This is incorrect, but will help some tests pass for now
+	  // TODO: remove this
+	  if (ref_node_id == UNKNOWN_NODEID)
+	    nr_ctx.lookup (expr.get_mappings ().get_nodeid ())
+	      .map (
+		[&ref_node_id] (NodeId resolved) { ref_node_id = resolved; });
```

Comment:
I think I would rather spend some time fixing the root cause for this. This seems a bit dodgy. what do you think is missing from nr2.0 in order to enable this? simply resolving each segment when we resolve a path?

Reply:
do your lookup on the ast_node_it which is the segment not the _path_ name resolution cant resolve a full path on its own.